lib/ex_format.ex

```diff
@@ -354,6 +354,17 @@ defmodule ExFormat do
   end
   defp parenless_capture?(_), do: false
 
+  defp handle_left_spec_op({target, meta, args} = left, state) do
+    cond do
+      Map.get(state, :type_spec?) == true and args == [] ->
+        {target, meta, nil}
+      Map.get(state, :fun_spec?) == true and args == nil ->
```

Comment:
Do the same as the other PR here. I would do this when initializing the state:

```elixir
state = %{
  ...,
  in_spec: nil,
}
```

and have `:in_spec` be either `:type` or `:def`. Then you can do

```elixir
case state.in_spec do
  :type when args == [] ->
    ...
  :def when is_nil(args) ->
    ...
  nil ->
    left
end
```